src/diffpy/structure/lattice.py

```diff
@@ -271,34 +370,75 @@ def setLatBase(self, base):
 
     def abcABG(self):
         """Return a tuple of 6 lattice parameters.
+
+        Returns
+        -------
+        rv : tuple
+            tuple of (a, b, c, alpha, beta, gamma)
         """
         rv = (self.a, self.b, self.c, self.alpha, self.beta, self.gamma)
         return rv
 
 
     def reciprocal(self):
-        """Return the reciprocal lattice to self.
+        """Return the reciprocal lattice.
+
+        Returns
+        -------
+        rv : Lattice
+            Reciprocal lattice of current lattice.
         """
         rv = Lattice(base=numpy.transpose(self.recbase))
         return rv
 
 
     def cartesian(self, u):
         """Return Cartesian coordinates of a lattice vector.
+
+        Parameters
+        ----------
+        u : ndarray
+            A lattice vector being projected to.
+
+        Returns
+        -------
+        rc : ndarray
+            Cartesian coordinates of a lattice vector.
         """
         rc = numpy.dot(u, self.base)
         return rc
 
 
     def fractional(self, rc):
         """Return fractional coordinates of a Cartesian vector.
+
+        Parameters
+        ----------
+        rc : ndarray
+            A lattice vector being projected to.
```

Comment:
Here `rc` is a vector in Cartesian coordinates or an N-by-3 array of Cartesian vectors.